dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/DubboShutdownHook.java

```diff
@@ -85,6 +85,8 @@ public void doDestroy() {
         if (!destroyed.compareAndSet(false, true)) {
             return;
         }
+        // unregister the shutdownHook
+        unregister();
```

Comment:
don't call `unregister` here. When shutdown hook runs, it will lead to IllegalStateException if we try to remove the hook.

```java
    static synchronized void add(Thread hook) {
        if(hooks == null)
            throw new IllegalStateException("Shutdown in progress");

        if (hook.isAlive())
            throw new IllegalArgumentException("Hook already running");

        if (hooks.containsKey(hook))
            throw new IllegalArgumentException("Hook previously registered");

        hooks.put(hook, hook);
    }
```

Reply:
> don't call `unregister` here. When shutdown hook runs, it will lead to IllegalStateException if we try to remove the hook.
> 
> ```java
>     static synchronized void add(Thread hook) {
>         if(hooks == null)
>             throw new IllegalStateException("Shutdown in progress");
> 
>         if (hook.isAlive())
>             throw new IllegalArgumentException("Hook already running");
> 
>         if (hooks.containsKey(hook))
>             throw new IllegalArgumentException("Hook previously registered");
> 
>         hooks.put(hook, hook);
>     }
> ```

I remove it